db/migrate/20141110084848_fix_puppetclass_total_hosts.rb

```diff
@@ -1,6 +1,11 @@
 class FixPuppetclassTotalHosts < ActiveRecord::Migration
   def up
-    Rake::Task['puppet:fix_total_hosts'].invoke
+    if User.anonymous_admin(false) && Puppetclass.count > 0
+      User.current = User.anonymous_admin
+      Puppetclass.all.each(&:update_total_hosts)
+    else
```

Comment:
Sure but it will end up in the Rails log. I think it's kinda useless. Do you happen to know how to display a nice message to the user?